client/state/signup/progress/actions.js

```diff
@@ -45,11 +45,15 @@ function recordSubmitStep( stepName, providedDependencies ) {
 			}
 
 			// Ensure we don't capture identifiable user data we don't need.
-			if ( includes( [ 'email', 'address', 'phone' ], propName ) ) {
+			if ( includes( [ 'email' ], propName ) ) {
 				propName = `user_entered_${ propName }`;
 				propValue = !! propValue;
 			}
 
+			if ( propName === 'cart_item' && typeof propValue !== 'string' ) {
+				propValue = toPairs( propValue ).join( ';' );
```

Comment:
I spent some time trying to break the implicit use of `Array.prototype.toString()` here but I could not. I was wondering if in non-english locales it'd use a seperator other than comma, but it seems to be consistent. I even read the ECMAScript standard to check this behaviour was standardised 😅 

Still, this seems too magical for me personally. I think spelling it out explicitly is better documentation for when it comes time to analyse this event data.

```js
propValue = toPairs( propValue )
  .map( pair => pair.join( ',' ) )
  .join( ';' );
```

Reply:
I agree. I let the implementation of `Object.entries` determine the data structure, which probably is a lazy way anyway. 

Using your suggestions, I could define the separators explicitly, and send something more useful (and parseable), e.g., `key1:value1,key2:value2` instead of `key1,value1;key2,value2`